contracts/contracts/SNStateProofVerifier.sol

```diff
@@ -123,6 +124,27 @@ contract SNStateProofVerifier is
         return hashvalue;
     }
 
+    // based on https://docs.starknet.io/documentation/architecture_and_concepts/Hashing/hash-functions/
+    function poseidonHashMany(
+        uint256[] memory elems
+    ) public view returns (uint256) {
+        uint256[3] memory state = [uint256(0), uint256(0), uint256(0)];
```

Comment:
maybe I'm missing something, but I feel that the code here never updates `state[2]`.

Reply:
oh, it's probably updated by the function `poseidon.poseidon(state)`, right?

Reply:
Yes that is correct